PYME/resources/web/static/recipe_editor.js

```diff
@@ -0,0 +1,119 @@
+
+condense=function(y){
+    var f = Math.floor(y);
+    var g = y-f;
+
+    return 5/8*(f/2) + g/4;
+};
+
+const zip = (arr2, arr1) => arr1.map((k, i) => [k, condense(arr2[i])]);
+const input_type_mappings = {string : 'text', number : 'number', boolean : 'checkbox'};
+
+var vm = undefined;
+// jigna.models are ready only when the deferred called ready is done.
+Vue.config.devtools = true;
+//Vue.config.debug = true;
+
+const cols = ['turquoise', 'emerald', 'peter-river', 'amethyst', 'wet-asphalt', 'sunflower', 'carrot', 'alizarin'];
+
+
+
+Vue.component('trait-input',{
+   props: ['value', 'type'],
+   template: `
+        <input :value="value" :checked="value" :type="input_type(type)" v-on:input="$emit('input', format_value($event))">`,
+   methods: {
+       input_type: function(type){return input_type_mappings[type];},
+       format_value: function(event){
+           var val = event.target.value;
+           switch(this.type){
+               case 'number':
+                   return parseFloat(val);
+               case 'boolean':
+                   val = event.target.checked;
+                   console.log(val, typeof(val));
+                   return val;
+               default:
+                   return val;
```

Comment:
If I understand this right @David-Baddeley we'll still be able to use File/Enum types, there just won't be any selectors for either of them, correct? (and I guess the enum's will have to be str which I think they usually are)

Reply:
I haven't pulled/fiddled with this, since I think I should work on a couple other things first, but would it be crazy to check if val startswith `{USERFILE` and if so call `select_file`?

Reply:
Yep - anything other than number or boolean gets a string editor. Should be fine, if inconvenient, for `File` and `Enum`. Probably won't work for any of the `List`, `Dict` etc ... traits.

I'd also thought about doing a check for `"USERFILE"`, but would want to know that we can't do things properly for all file inputs instead first. Suspect it might only be marginally more work to get something that works for all `File` inputs and, if possible that would be vastly preferable.